### DISCUSSION TITLE
Using two or more independent sensor networks: Change the MAC_PREFIX or use separate reading ids?

### DISCUSSION THREAD

**Discussion by shaffenmeister** [timmbogner/Farm-Data-Relay-System#174] [Q&A] [answered]
Hi all,

I would like to operate _two or more independent sensor networks_ in the same vicinity.
If I understand correctly I can reach this goal using
1. different READING IDS / UNIT_MACS for all nodes/gateways or by
2. changing the MAC_PREFIX for each network.

Wireless "congestion" will not be an issue, the number of nodes will not exceed 10 nodes per network.

What is the preferred route to go?
Using different reading ids is easy and straightforward.
Changing the MAC_PREFIX requires a modification of src/fdrs_globals.h. Then I would need to maintain a branch of FDRS for each network.

Any suggestions?

Thank you in advance!
And thanks a lot for this diamond (to cite the guy with the Swiss accent... ;-) )

  Christoph

**Comment by timmbogner** [comment, marked as answer]
Great question! I think both should work equally, although I haven't tested option 2 as much. What it comes down to then, is which would be easier for you?

 If both sensor systems are nearly identical, it might make sense to keep two identical copies of the system files and just change the MAC prefix. If you are feeding all of your data into the same database and it would be better for your sensor IDs to be unique, that's a good case for the first option.
 
 At this point there's no preferred method, but I'd love to hear about what you choose and how it goes!
 
 As a note:
 With LoRa there is a  syncword that should differentiate between networks, although that is also yet untested. I think if LoRa were in the mix I might steer you more towards option 1.

**Reply by shaffenmeister** [reply]
Thanks for elaborating on this! You are adding even more things to consider. ;-)

I am using a very basic set-up for both networks - ESP NOW nodes only and rather self-contained ESP32 gateways with no external connections.

I decided on going the MAC_PREFIX route to keep both systems separate and as such being able to freely choose sensor ids.

To avoid maintaining multiple branches I created my own branch only modifying `src/fdrs_globals.h` as such:

`#ifndef FDRS_GLOBALS`
`#define FDRS_GLOBALS`
`...`
`#endif`

Now I can use a local include file like `myglobals.h` which replaces `fdrs_globals.h` by using `#define FDRS_GLOBALS` accordingly.

Hope that makes sense.

**Reply by timmbogner** [reply]
Awesome! That ```#ifndef``` actually used to be present, but I inexplicably removed it while refactoring a while back. If you want, you can add it back and submit the change as a pull request, otherwise I'll put it in later this week.

Thanks for the feedback!

**Reply by shaffenmeister** [reply]
Will create a pull requests, no worries...
